hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/NettyRpcServer.java

```diff
@@ -181,4 +204,80 @@ public int getNumOpenConnections() {
     // allChannels also contains the server channel, so exclude that from the count.
     return channelsCount > 0 ? channelsCount - 1 : channelsCount;
   }
+
+  private synchronized void initSSL(ChannelPipeline p, boolean supportPlaintext) throws
+    X509Exception {
+    SslContext nettySslContext;
+
+    SSLContextAndOptions sslContextAndOptions = x509Util.getDefaultSSLContextAndOptions();
+    nettySslContext = sslContextAndOptions
+      .createNettyJdkSslContext(sslContextAndOptions.getSSLContext(), false);
+
+    if (supportPlaintext) {
```

Comment:
[nit] does this `supportPlaintext` is always `true` ? should we just support the plain text mode? or if you would like to have a configuration for it.

Reply:
Good point. Yes, we should support it. Let me add a config option.

Reply:
Long term, we should know if there's an impact of the OptionalSslHandler being in the pipeline. Ideally, we could always set this handler as long as the necessary keystore/truststore user-provided configuration values are given.

Reply:
Thoughts on just allowing binding to 2 ports instead? I realize this is mirrored after Zookeeper, but I think for hbase using 2 ports would be:

1. More performant (does not need to do the pipeline replacements used in OptionalSslMode). this may not matter so much since it's only on initial connection, but worth mentioning.
2. More flexibile. One could listen SSL on one port, and plaintext on another. The plaintext port could be locked down with network rules, allowing only intra-cluster plaintext.

Otherwise I anticipate wanting to add functionality to limit plaintext mode by ip address or subnet.